otter/convergence/steps.py

```diff
@@ -350,18 +341,22 @@ def report_api_failure(result):
             Otherwise, retry.
             """
             err_type, error, traceback = result
-            if error.code == 404:
-                return StepResult.FAILURE, [
-                    ErrorReason.Exception(result)]
-            if error.code == 422:
-                message = try_json_with_keys(error.body, ("message",))
-                if message and _CLB_DELETED_PATTERN.search(message):
-                    return StepResult.FAILURE, [
-                        ErrorReason.Exception(result)]
-            return StepResult.RETRY, [ErrorReason.Exception(result)]
+            status = StepResult.FAILURE
+
+            known_retryable = err_type in (CLBDuplicateNodesError,
```

Comment:
I think maybe with a few small abstractions we could simplify this function a lot, but that's just a side thought. LGTM :)